src/components/DatePicker/DatePicker.tsx

```diff
@@ -310,7 +322,7 @@ export const DatePicker = ({
         onFocus={e => (e.target.readOnly = true)}
         onInputClick={() => {
           if (pickerRef.current && !isDesktop) {
-            pickerRef.current.setOpen(!pickerRef.current.isCalendarOpen)
+            pickerRef.current.setOpen(!pickerRef.current.isCalendarOpen())
```

Comment:
How was this working before? Either it was broken or I'm missing some knowledge.

Reply:
The types for `react-datepicker` are awful. They only ported to a typescript approach in the latest version 7.X (and we're on 6.9 with an external types library).

It is actually a method, not a property, so it should be called.

This case can't even be reached on a mobile view due to some dark magic with the `onFocus` handler above ([reference to source code](https://github.com/Hacker0x01/react-datepicker/blob/02ce53f2c50ae77b177da1021b6e7f2e6faaaf0f/src/index.jsx#L484)). Still, the end-result is absolutely fine.